### PR TITLE
Adding deep water approximations

### PR DESCRIPTION
**Pull request by b0ndman**
Deep water approximations for wave celerity (so sinh doesn't blow up to infinity) using new functions wave_length() and depth_regime(). wave_length input is wave number array, output is array of wavelengths indexed by frequency. depth_regime inputs are wavelength array, water column depth, and optional defining ratio of water depth/wavelength.

Deep water approximation for energy_flux also implemented using IEC 62600-100 eq. 8, resolves TODO in that function

### CODE REVIEW

**Comment by ssolson** [comment]
@b0ndman thank you for your contribution to MHKiT this looks like some really great functionality! I will be out of the office and unable to perform a formal review until later this month. The only thing currently missing is the tests for the new functions and possibly adding a test for deep=True in the current wave celerity test. If you are open to writing a couple of quick tests I could throw them into the test modules or we can work on them when I get back.

@rpauly18 speaking of tests it looks like NREL-rex failed the CI test on this PR build. If you get a moment could you check that out?

Thanks again @b0ndman  great stuff!

**Comment by rpauly18** [comment]
@ssolson yes, I have seen that and have started looking into it. 

@b0ndman thanks for the contribution! I am also here to help with developing tests, just let us know if you are open to working on them.

**Comment by b0ndman** [comment]
Apologies for kinda disappearing after doing the pull request (I'm on a bit of a tight work schedule, with the work on this stemming from that). I am open to helping out, at a limited capacity, and have noticed the NREL-rex failing also.

**Comment by rpauly18** [comment]
@b0ndman thanks for offering to help. I have added a couple of tests for the depth_regime and wave_length functions, but if you have time and ability to review these tests and possibly add one for the deep water flag in energy flux it would be much appreciated. I am actively working on the NREL-rex issues that are being triggered by too many requests to the WPTO hindcast data server.

**Review by ssolson — COMMENTED** [review]
Hey, Rebecca apologizes for the delay of this review. Overall I had a couple of questions before we proceed. We can also schedule a chat to talk them over. Were you able to push directly to bondman's master branch to make changes?

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,9 @@
 $ cat .gitignore
 
+# scratch file
+scratch.py
+scratch_data.txt
```
[the remark this answers is not in the document]
Comment (ssolson) [reply .gitignore]: removed in bc2720f

File: mhkit/wave/resource.py [hunk 2]
```diff
@@ -575,31 +577,43 @@ def energy_flux(S, h, rho=1025, g=9.80665):
     # TODO: Add deep water flag
```
[the remark this answers is not in the document]
Comment (ssolson) [reply mhkit/wave/resource.py:577]: Removed in ac13e8e. Also switched logic for better readability.

File: mhkit/wave/resource.py [hunk 3]
```diff
@@ -621,16 +637,64 @@ def wave_celerity(k, h, g=9.80665):
...
+    To compute: 2*pi/wavenumber
+
+    Parameters
+    -------------
+    k: pandas Dataframe
+        Wave number [1/m] indexed by frequency
+
+    Returns
+    ---------
+    l: pandas Dataframe
```
[the remark this answers is not in the document]
Comment (ssolson) [reply mhkit/wave/resource.py:689]: Variable can't be `lamda` because it's a built-in function. I will stick with "l" I guess.

**Comment by ssolson** [comment]
I have addressed all of my comments above. I think there are 2 issues outstanding after working with the code more.

1. Cannot change `depth_ratio`  in `depth_regime` when calling from `wave_celerity`
2. No tests for the deep=True, and depth_check_=True

Currently, depth_regime allows a user to specify a depth ratio for the depth regime. However, when using wave celerity there is no way to change the default value of 2. To address this I suggest instead of only a boolean for the parameter `deep` we also allow ints/ floats. If an int/float is passed this is used instead of the default value.

For point 2 I will add tests to check the other half of the `wave_celerity` and `energy_flux` functions

File: mhkit/wave/resource.py [hunk 4]
```diff
@@ -621,16 +637,64 @@ def wave_celerity(k, h, g=9.80665):
...
 
+def wave_length(k):
+    """
+    Calculates wave length from wave number
+    To compute: 2*pi/wavenumber
+
+    Parameters
+    -------------
+    k: pandas Dataframe
+        Wave number [1/m] indexed by frequency
```
[the remark this answers is not in the document]
Comment (rpauly18) [reply mhkit/wave/resource.py:684]: Frequency is needed to calculate wave number, so when calculating it you get a k indexed by frequency. I suppose there could be use cases where a user would use a k not directly calculated from the MHKiT wave_number function that is a wave number for each time. So, for this particular function I suppose the index does not matter, I think he was just trying to be consistent with other functions that have k.

**Comment by rpauly18** [comment]
Good catch on point 1. I think it would be better to add ratio as an optional parameter in wave_celerity with the default=2. We can in the doc string indicate that for the parameter to have any effect check_depth must equal True. I will add this.

**Comment by ssolson** [comment]
@rpauly18 I addressed the missing test and coverage has been raised by 1%. I believe this is ready to go if you have a chance to look at it,

**Pull request merged**